File: WeatheriOS/Weather/Models/WeatherData.swift
//
//  WeatherData.swift
//  Weather
//
//  Created by Rick Pasetto on 12/21/25.
//

import Foundation

struct WeatherInfo: Identifiable {
    let id = UUID()
    let city: City
    let currentTemperature: Int?
    let currentCondition: String?
    let forecast: [ForecastPeriod]
}

struct ForecastPeriod: Identifiable {
    let id = UUID()
    let name: String
    let temperature: Int
    let shortForecast: String
}

struct ForecastPeriodDTO: Codable {
    let name: String
    let temperature: Int
    let shortForecast: String
    
    func toForecastPeriod() -> ForecastPeriod {
        ForecastPeriod(name: name, temperature: temperature, shortForecast: shortForecast)
    }
}

struct PointsResponse: Codable {
    let properties: PointsProperties
}

struct PointsProperties: Codable {
    let gridId: String
    let gridX: Int
    let gridY: Int
    let forecast: URL
}

struct ForecastResponse: Codable {
    let properties: ForecastProperties
}

struct ForecastProperties: Codable {
    let periods: [ForecastPeriodDTO]
    
    var forecastPeriods: [ForecastPeriod] {
        periods.map { $0.toForecastPeriod() }
    }
}

struct ObservationResponse: Codable {
    let properties: ObservationProperties?
}

struct ObservationProperties: Codable {
    let temperature: TemperatureValue?
    let textDescription: String?
}

struct TemperatureValue: Codable {
    let value: Double?
}

File: WeatherAndroid/app/src/main/java/com/weatherdemo/models/WeatherData.kt
package com.weatherdemo.models

import java.util.UUID

data class WeatherInfo(
    val id: String = UUID.randomUUID().toString(),
    val city: City,
    val currentTemperature: Int?,
    val currentCondition: String?,
    val forecast: List<ForecastPeriod>
)

data class ForecastPeriod(
    val id: String = UUID.randomUUID().toString(),
    val name: String,
    val temperature: Int,
    val shortForecast: String
)

data class ForecastPeriodDTO(
    val name: String,
    val temperature: Int,
    val shortForecast: String
) {
    fun toForecastPeriod(): ForecastPeriod {
        return ForecastPeriod(name = name, temperature = temperature, shortForecast = shortForecast)
    }
}

data class PointsResponse(
    val properties: PointsProperties
)

data class PointsProperties(
    val gridId: String,
    val gridX: Int,
    val gridY: Int,
    val forecast: String
)

data class ForecastResponse(
    val properties: ForecastProperties
)

data class ForecastProperties(
    val periods: List<ForecastPeriodDTO>
) {
    val forecastPeriods: List<ForecastPeriod>
        get() = periods.map { it.toForecastPeriod() }
}

data class ObservationResponse(
    val properties: ObservationProperties?
)

data class ObservationProperties(
    val temperature: TemperatureValue?,
    val textDescription: String?
)

data class TemperatureValue(
    val value: Double?
)

data class StationsResponse(
    val features: List<StationFeature>
)

data class StationFeature(
    val properties: StationProperties
)

data class StationProperties(
    val stationIdentifier: String
)

